checks.d/sequins.py

```diff
@@ -0,0 +1,77 @@
+import requests
+
+from checks import AgentCheck
+
+
+class Sequins(AgentCheck):
+    DEFAULT_TIMEOUT = 1
+    DEFAULT_MAX_DBS = 100
+    CONNECT_CHECK_NAME = 'sequins.can_connect'
+
+    def check(self, instance):
+        if 'url' not in instance:
+            self.log.info("Skipping instance, no url found.")
+            return
+
+        instance_tags = instance.get('tags', [])
+        max_dbs = instance.get('max_dbs', self.DEFAULT_MAX_DBS)
+        timeout = self.init_config.get('default_timeout', self.DEFAULT_TIMEOUT)
+        resp = self.get_json(instance['url'], timeout)
+
+        self.gauge('sequins.db_count', len(resp['dbs']), tags=instance_tags)
+        for db_name, db in resp['dbs'].iteritems():
+            db_tags = instance_tags + ['sequins_db:%s' % db_name]
+
+            num_dbs = len(db['versions'])
+            if num_dbs > max_dbs:
+                raise Exception("%d dbs is more than the configured maximum (%d)" % (num_dbs, max_dbs))
+
+            self.gauge('sequins.version_count', num_dbs, db_tags)
+
+            for version_name, version in db['versions'].iteritems():
+                version_tags = db_tags + ['sequins_version:%s' % version_name]
+                self.gauge('sequins.partition_count', version['num_partitions'], version_tags)
+                self.gauge('sequins.missing_partition_count', version['missing_partitions'], version_tags)
+                self.gauge('sequins.underreplicated_partition_count', version['underreplicated_partitions'], version_tags)
+                self.gauge('sequins.overreplicated_partition_count', version['overreplicated_partitions'], version_tags)
+                self.gauge('sequins.average_replication', version['average_replication'], version_tags)
+
+                node_counts = {}
+                for node in version['nodes'].itervalues():
+                    st = node['state']
+                    node_counts[st] = node_counts.get(st, 0) + 1
+
+                self.gauge('sequins.node_count', len(version['nodes']), version_tags)
+                self.gauge('sequins.building_node_count', node_counts.get('BUILDING', 0), version_tags)
```

Comment:
Per your comments above, I think it's a good idea to make these tags. It seems like it would be a common desire to see all the of the states at once without having to add each specific metric to the chart, plus it looks like these sum up logically to the total number of nodes. So that's two good reasons imo to follow you instinct to use tags!